services/waf/src/it/java/software/amazon/awssdk/services/waf/WafIntegrationTest.java

```diff
@@ -67,11 +68,9 @@ public static void tearDown() throws IOException {
 
     private static void deleteIpSet() {
         if (ipSetId != null) {
-            final String changeToken = newChangeToken();
-            client.deleteIPSet(DeleteIPSetRequest.builder()
-                    .ipSetId(ipSetId)
-                    .changeToken(changeToken)
-                    .build());
+            Waiter.run(() -> client.deleteIPSet(r -> r.ipSetId(ipSetId).changeToken(newChangeToken())))
+                  .ignoring(WAFNonEmptyEntityException.class)
```

Comment:
Does this just run the operation until there's no `WAFNonEmptyEntityException` exception?

Reply:
Yes.